### ISSUE TITLE
Godot 3.2.2 assert regression

### ISSUE THREAD

**Issue by allkhor** [godotengine/godot#38270]
**Godot version:**
Godot 3.2.2 78f0cf4
**OS/device including version:**
Xubuntu 16.04.6
**Issue description:**
This code now throw error:
```
extends Node

func _ready():
	assert(test()) # Parse Error: The method "test" isn't declared in the current class.

func test() -> bool:
	"""I'm a simple test case."""
	return true
 
```
Assert regression from 8dc8833.
**Steps to reproduce:**
Open minimal reproduction project.
**Minimal reproduction project:**
[assert.zip](https://github.com/godotengine/godot/files/4541522/assert.zip)

**Comment by akien-mga** [comment]
CC @BigRed-118 @vnen

**Comment by BigRed-118** [comment]
Can confirm, removing the recently added line from the case GDScriptTokenizer::TK_PR_ASSERT in gdscript_parser.cpp makes this go away. Looking into it now.

```cpp
_reduce_node_type(args[0]);
```

**Comment by BigRed-118** [comment]
Issue stems from when _reduce_node_type is called from GDScriptParser::_parse_block for the assert, the current_class's scope is within the current function, so it doesn't have test in the class_type.functions. Putting the check in GDScriptParser::_check_block_types (which is where I should have placed it in the first place, with the Node::TYPE_ASSERT being cased out but nothing done for it), keeps the assert safe check functionality and no longer gives the above error. Now that we're setting the AssertNode line correctly from the original change, we now have the correct reference to use here.

```cpp
case Node::TYPE_ASSERT: {
    AssertNode *an = static_cast<AssertNode *>(statement);
    _mark_line_as_safe(an->line);
    _reduce_node_type(an->condition);
    _reduce_node_type(an->message);
} break;
```

Also, it looks like the optional second argument for assert message wasn't being checked before. I went ahead and added it to the check above as you can see. Which produces the following:
![image](https://user-images.githubusercontent.com/52541536/80433589-f393cb80-88bc-11ea-98e4-383e281d2e2c.png)
![image](https://user-images.githubusercontent.com/52541536/80433573-eaa2fa00-88bc-11ea-9976-436b386a3642.png)
Then with a normal string input:
![image](https://user-images.githubusercontent.com/52541536/80433689-30f85900-88bd-11ea-8438-12d94fab7c59.png)


Previously, there was no sign of an error:
![image](https://user-images.githubusercontent.com/52541536/80433454-94ce5200-88bc-11ea-924f-ad6ee8c01e08.png)


Is that something that'd be alright to add in? I'll open a pull request for this fix. If anyone has any concerns or issues with this let me know. Thank you @allkhor for reporting.

**Comment by vnen** [comment]
Oh, I didn't realize it was adding a `reduce` call on the first parser pass. That's why it broke.

**Comment by akien-mga** [comment]
Fixed by #38279.
